File: spring/tomcat-monitoring-example/src/main/java/com/example/tomcatmonitoringexample/MyController.java
package com.example.tomcatmonitoringexample;

import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class MyController {

    @GetMapping
    Map<String, String> index() {
        return Map.of("result", "ok");
    }
}

File: spring/mapstruct-spring-extensions-kotlin/src/main/kotlin/com/example/mapstruct_spring_extensions_kotlin/MyController.kt
package com.example.mapstruct_spring_extensions_kotlin

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class MyController(
    private val carMapper: CarMapper,
) {
    private val log: Logger = LoggerFactory.getLogger(MyController::class.java)

    @PostMapping
    fun index(@RequestBody car: Car): CarDto {
        log.info("car: {}", car)

        val dto = carMapper.convert(car)

        log.info("dto: {}", dto)

        return dto
    }

}
